3rdparty/mbedtls/mbedtls_hardware_poll.c

```diff
@@ -19,13 +93,48 @@ int mbedtls_hardware_poll(
     size_t* olen)
 {
     int ret = -1;
+    oe_entropy_kind_t kind = OE_ENTROPY_KIND_NONE;
     OE_UNUSED(data);
 
     if (olen)
         *olen = 0;
 
-    if (oe_get_entropy(output, len) != OE_OK)
+    if (oe_get_entropy(output, len, &kind) != OE_OK)
+        goto done;
+
+    if (kind == OE_ENTROPY_KIND_RDSEED || kind == OE_ENTROPY_KIND_OPTEE)
+    {
+        /* According to Intel's DRNG software implementation guide, RDSEED
+         * produces values that are already passed through a conditioner that
+         * hashes pairs of 256-bit raw entropy samples via AES-CBC-MAC, so no
+         * further work needs to be done.
+         *
+         * For OPTEE TEE_GenerateRandom, the actual predictive resistance of
+         * underlying implementation may vary, so this simply takes the value
+         * provided as is. */
+    }
+    else if (kind == OE_ENTROPY_KIND_RDRAND)
+    {
+        /* If RDSEED is not supported, fallback to using RDRAND to obtain a
+         * seed for entropy. */
+        unsigned char* p = (unsigned char*)output;
+        size_t bytes_left = len;
+        while (bytes_left > 0)
+        {
+            uint8_t* seed_bytes = NULL;
+            size_t seed_size = 0;
+
+            if (_get_seed_from_rdrand(&seed_bytes, &seed_size) != OE_OK)
```

Comment:
It woulde be better to avoid the double call to `oe_get_entropy`. In line 102 and the other via __get_seed_from_rdrand

Reply:
This is by design. It produces the most straightforward separation of layering between a crypto library and the `oe_get_entropy` method in the absence of hash functions embedded in oecore itself to condense entropy itself.